File: src/main/java/com/ekart/CartController.java
package com.ekart;

import com.ekart.model.AddressInfo;
import com.ekart.model.Card;
import com.ekart.service.ICartService;
import com.ekart.service.IPaymentService;
import com.ekart.service.IShipmentService;

public class CartController {

  private ICartService cartService;
  private IPaymentService paymentService;
  private IShipmentService shipmentService;

  public CartController(ICartService cartService, IPaymentService paymentService,
                        IShipmentService shipmentService) {
    this.cartService = cartService;
    this.paymentService = paymentService;
    this.shipmentService = shipmentService;
  }

  public String checkOut(Card card, AddressInfo addressInfo) {
    if(cartService.numberOfItems() == 0) {
      return "Cart Is Empty";
    }
    boolean result = paymentService.charge(cartService.total(), card);
    if(result) {
      cartService.clear();
      shipmentService.ship(addressInfo, cartService.items());
      return "Charged";
    } else {
      return "Not Charged";
    }
  }
}

File: src/main/.net/CartController.cs
using ShoppingApp.Services;

namespace ShoppingApp
{
    public class CartController
    {
        private readonly ICartService _cartService;
        private readonly IPaymentService _paymentService;
        private readonly IShipmentService _shipmentService;

        public CartController(ICartService cartService, IPaymentService paymentService, IShipmentService shipmentService)
        {
            _cartService = cartService;
            _paymentService = paymentService;
            _shipmentService = shipmentService;
        }

        public string CheckOut(Card card, AddressInfo addressInfo)
        {
            if (!_cartService.Items().Any())
                return "cart empty";

            var result = _paymentService.Charge(_cartService.Total(), card);
            if (result)
            {
                _shipmentService.Ship(addressInfo, _cartService.Items());
                _cartService.Clear();
                return "charged";
            }
            else
            {
                return "not charged";
            }
        }
    }
}